File: app/src/main/java/ai/rotor/rotorvehicle/ui/controller/ControllerViewModel.java
package ai.rotor.rotorvehicle.ui.controller;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ControllerViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public ControllerViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is the controller fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/ai/rotor/rotorvehicle/ui/controller/ControllerViewModel.kt
package ai.rotor.rotorvehicle.ui.controller

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class ControllerViewModel : ViewModel() {

    private val mText: MutableLiveData<String>

    val text: LiveData<String>
        get() = mText

    init {
        mText = MutableLiveData()
        mText.value = "This is the controller fragment"
    }
}